### DISCUSSION TITLE
Possibilty For Adding ratelimits

### DISCUSSION THREAD

**Discussion by MalikWhitten67** [pockethost/pockethost#279] [Q&A]
Currently stuck on how I should limit interactions to the Api on pocket host's side.

**Comment by benallfree** [comment]
Have you tried asking in the PocketBase forums? This seems better as a PocketBase feature (rate limiting). 

What trouble are you having?

**Reply by MalikWhitten67** [reply]
I wanted to possibly implement it via hooks but was unsure.

**Reply by benallfree** [reply]
Yes, but I mean what is the exact use case? Why are you needing to rate limit users?

**Reply by MalikWhitten67** [reply]
Well users are posting content and updating records as my app postr is a social media platform. With that there can be abuse.

**Reply by benallfree** [reply]
Rate limiting is one possible solution, but maybe not the best for cases like this. You probably have other business rules restricting who can post and how often. For example, you might want some middleware to filter duplicate posts, posts that are too short or too long, etc.

You could address these issues in a pre-commit hook, and that could be an interesting approach too, but ultimately still hard to maintain.

To me, the most flexible approach is to lock down the `posts` table to prevent inserts by any client. Then, make an API endpoint that validates the post (including rate limiting) before inserting. This makes it easy to re-use whatever business rules you create while keeping the attack surface as small as possible.

From a practical standpoint, I would absolute wait until you have an issue before working too much on security and validation. It is easy to burn up dev cycles that could be better spent creating features that the users care about and building your platform. If you get to a point where abuse is happening, it's easy to add rules to prevent it.

**Reply by MalikWhitten67** [reply]
Thats feasible I'll give it a try.
